sql/src/main/java/com/vmware/ddlog/DDlogJooqProvider.java

```diff
@@ -111,4 +161,72 @@ private Object structToValue(final Field<?> field, final DDlogRecord record) {
                 throw new RuntimeException("Unknown datatype %s of field %s in table %s in update received");
         }
     }
+
+    private DDlogRecord toDDlogRecord(final String tableName, final String[] args) {
+        final DDlogRecord[] recordsArray = new DDlogRecord[args.length];
+        final List<Field<?>> fields = tables.get(tableName.toUpperCase());
+
+        int fieldIndex = 0;
+        for (final Field<?> field : fields) {
+            final Class<?> cls = field.getType();
+            try {
+                // Handle nullable columns here
+                if (args[fieldIndex] == null) {
+                    recordsArray[fieldIndex] = DDlogRecord.makeStruct(DDLOG_NONE_TYPE, new DDlogRecord[0]);
+                }
+                else {
+                    switch (cls.getName()) {
+                        case BOOLEAN_TYPE:
+                            recordsArray[fieldIndex] = maybeOption(field,
```

Comment:
there's a bunch of common functionality between all these cases that could be factored out, not sure it's worth.

Reply:
Not sure I can though. There's no way to pattern match on the class names, because I don't know the schema in advance. And the way to parse the supplied SQL is different per type.

Reply:
Maybe this code will also get simplified if I use the presto parser here.